libqtile/widget/keyboardlayout.py

```diff
@@ -22,107 +22,274 @@
+            " layout.",
+        ),
+        (
+            'geometry',
+            None,
+            "Specifies the name of the geometry component used to"
+            " construct a keyboard layout.",
+        ),
+        (
+            'dirs',
+            None,
+            "Adds a directory to the list of directories to be used to"
+            " search for specified layout or rules files.",
+        ),
+        (
+            'keycodes',
+            None,
+            "Specifies the name of the keycodes component used to"
+            " construct a keyboard layout.",
+        ),
+        (
+            'keymap',
+            None,
+            "Specifies the name of the keymap description used to"
+            " construct a keyboard layout.",
+        ),
+        (
+            'layout_groups',
+            None,
+            "Specifies the name of the layout used to determine the"
+            " components which make up the keyboard description. For"
+            " example: ['us', 'ru', 'sk'], 'us,ru,sk'.",
+        ),
+        (
+            'model',
+            None,
+            "Specifies the name of the keyboard model used to determine"
+            " the components which make up the keyboard description."
+            " For example: 'pc101', 'pc105'.",
+        ),
+        (
+            'option',
+            None,
+            "Specifies the name of an option to determine the"
+            " components which make up the keyboard description. For"
+            " example: ['compose:menu', 'grp_led:scroll'],"
+            " 'compose:menu,grp_led:scroll'.",
+        ),
+        (
+            'rules',
+            None,
+            "Specifies the name of the rules file used to resolve the"
+            " requested layout and model to a set of component names.",
+        ),
+        (
+            'symbols',
+            None,
+            "Specifies the name of the symbols component used to"
+            " construct a keyboard layout.",
+        ),
+        (
+            'synch',
+            False,
+            "Force synchronization for X requests.",
+        ),
+        (
+            'types',
+            None,
+            "Specifies the name of the types component used to"
+            " construct a keyboard layout.",
+        ),
+        (
+            'variant',
+            None,
+            "Specifies which variant of the keyboard layout should be"
+            " used to determine the components which make up the"
+            " keyboard description.",
+        ),
     ]
 
-    def __init__(self, **config):
-        base.InLoopPollText.__init__(self, **config)
+    def __init__(self, **config) -> None:
+        base._TextBox.__init__(self, **config)
         self.add_defaults(KeyboardLayout.defaults)
+        self.raise_for_dependencies()
+        self.use_optional_dependencies_if_available()
+        event_loop = asyncio.get_event_loop()
+        event_loop.create_task(self.loop())
```

Comment:
Is there a reason to switch to this vs. just continuing to use InLoopPollText? Maybe we should just switch InLoopPollText to use this?

Reply:
Okay, there is no particular reason to switch to `_TextBox`.